File: nv_react_native_sdk_app/ios/NVSDKConfigModule.m
//
//  NVSDKConfigModule.m
//  nv_react_native_sdk_app
//
//  Created by Notifyvisitors Macbook Pro 001  on 10/03/26.
//

#import <Foundation/Foundation.h>

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(NVSDKConfig, NSObject)

RCT_EXTERN_METHOD(
  getNVBrandID:(RCTPromiseResolveBlock)resolve
  rejecter:(RCTPromiseRejectBlock)reject
)

@end

File: nv_react_native_sdk_app/ios/NVSDKConfigModule.swift
//
//  NVSDKConfigModule.swift
//  nv_react_native_sdk_app
//
//  Created by Notifyvisitors Macbook Pro 001  on 10/03/26.
//

import Foundation
import React
import ReactAppDependencyProvider

@objc(NVSDKConfig)
class NVSDKConfig: NSObject {

  @objc
  func getNVBrandID(_ resolve: RCTPromiseResolveBlock,
                 rejecter reject: RCTPromiseRejectBlock) {
    print("NVSDKConfig: getNVBrandID()")
    if let nvBrnadIDStr = Bundle.main.object(
           forInfoDictionaryKey: "nvBrandID"
       ) as? String {

           resolve(nvBrnadIDStr)
       } else {
           reject("NVSDKConfig", "BrandID not found", nil)
       }
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
